Engine/Component/ComponentMeshRenderer.cpp

```diff
@@ -185,4 +163,28 @@ void ComponentMeshRenderer::Copy(Component* component_to_copy) const
 {
 	*component_to_copy = *this;
 	*static_cast<ComponentMeshRenderer*>(component_to_copy) = *this;
-};
\ No newline at end of file
+};
+
+
+void ComponentMeshRenderer::SetMesh(uint32_t mesh_uuid)
+{
+	this->mesh_uuid = mesh_uuid;
+	if (mesh_uuid != 0)
+	{
+		this->mesh_to_render = std::static_pointer_cast<Mesh>(App->resources->Load(mesh_uuid));
+		owner->aabb.GenerateBoundingBox();
+	}
+}
+
+void ComponentMeshRenderer::SetMaterial(uint32_t material_uuid)
+{
+	this->material_uuid = material_uuid;
+	if (material_uuid != 0)
+	{
+		material_to_render = std::static_pointer_cast<Material>(App->resources->Load(material_uuid));;
```

Comment:
Why removing the template and adding an annoying switch if at the end of the day you need to cast it? I can't see the point of changing it here.

Reply:
yeah, that's right